tracer/src/Datadog.Trace/Ci/CIVisibility.cs

```diff
@@ -702,5 +694,37 @@ void CallBack(AgentConfiguration aConfiguration)
 
             return EventPlatformProxySupport.None;
         }
+
+        private class DiscoveryAgentConfigurationCallback
+        {
+            private readonly ManualResetEventSlim _manualResetEventSlim;
+            private readonly Action<AgentConfiguration> _callback;
+            private readonly IDiscoveryService _discoveryService;
+            private AgentConfiguration? _agentConfiguration;
+
+            public DiscoveryAgentConfigurationCallback(IDiscoveryService discoveryService)
+            {
+                _manualResetEventSlim = new ManualResetEventSlim();
+                LifetimeManager.Instance.AddShutdownTask(() => _manualResetEventSlim.Set());
+                _discoveryService = discoveryService;
+                _callback = CallBack;
+                _agentConfiguration = null;
+                _discoveryService.SubscribeToChanges(_callback);
+            }
+
+            public AgentConfiguration? WaitAndGet(int timeoutInMs = 5_000)
+            {
+                _manualResetEventSlim.Wait(timeoutInMs);
+                return _agentConfiguration;
+            }
+
+            private void CallBack(AgentConfiguration agentConfiguration)
+            {
+                _agentConfiguration = agentConfiguration;
+                _manualResetEventSlim.Set();
+                _discoveryService.RemoveSubscription(_callback);
+                Log.Debug("Agent configuration received.");
```

Comment:
Don't know if matters to you that this captures the `Log` from the parent class instead of `DiscoveryAgentConfigurationCallback`? 🤷‍♂️

Reply:
In this case it doesn't matter because the parent class is static, so there's no `this` to capture.